### ISSUE TITLE
Fine-tuning `contrast-agnostic` trained model for `EPI` data

### ISSUE THREAD

**Issue by rohanbanerjee** [sct-pipeline/contrast-agnostic-softseg-spinalcord#83]
Opening this issue to discuss the fine-tuning of the `contrast-agnositic` final trained model on the `EPI` data. 

Location of the trained `contrast-agnostic` model checkpoint: `duke/temp/muena/contrast-agnostic/final_monai_model/nnunet_nf=32_DS=1_opt=adam_lr=0.001_AdapW_CCrop_bs=2_64x192x320_20230918-2253`

Currently the `main.py` script under the `monai` folder in the repository does not have the functionality of loading the weights from checkpoint and loads the runs from the `wands` run. 

I will keep updating this issue with further details.

**Comment by valosekj** [comment]
Tagging @plbenveniste -- we applied the `contrast-agnostic` model on canproco PSIR/STIR images (context [here](https://github.com/ivadomed/canproco/issues/46)).

**Comment by naga-karthik** [comment]
Just to clarify @valosekj, the problem is here that of fine-tuning the contrast-agnostic model i.e. using the weights of the contrast-agnostic model as initialization for (re)training a _new_ model on Rohan's EPI data. 

The script that you show just runs inference using the contrast-agnostic model (which is _not_ what we're trying to do)

**Comment by valosekj** [comment]
> Just to clarify @valosekj, the problem is here that of fine-tuning the contrast-agnostic model i.e. using the weights of the contrast-agnostic model as initialization for (re)training a _new_ model on Rohan's EPI data.
> 
> The script that you show just runs inference using the contrast-agnostic model (which is _not_ what we're trying to do)

Correct, we ran the inference using the contrast-agnostic model to obtain GT for PSIR/STIR. Now, the GT could be used to fine-tune the contrast-agnostic model for these two contrasts.

**Comment by naga-karthik** [comment]
Ah okay, thanks for clarifying! I misunderstood your comment earlier -- editing it for clarity!
